pkg/server/definitionHandler.go

```diff
@@ -0,0 +1,218 @@
+package server
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"go.etcd.io/etcd/pkg/fileutil"
+
+	mycue "github.com/oam-dev/kubevela/pkg/cue"
+	"github.com/oam-dev/kubevela/pkg/server/util"
+	"github.com/oam-dev/kubevela/pkg/utils/common"
+	"github.com/oam-dev/kubevela/pkg/utils/system"
+)
+
+const (
+	// OpenAPISchemaDir is the folder name under ~/.vela/capabilities
+	OpenAPISchemaDir = "openapi"
+	// UsageTag is usage comment annotation
+	UsageTag = "+usage="
+	// ShortTag is the short alias annotation
+	ShortTag = "\\n+short"
+)
+
+// OpenAPISchema is the struct for OpenAPI Schema generated by Cue OpenAPI
+type OpenAPISchema struct {
+	OpenAPI    string     `json:"openapi"`
+	Components Components `json:"components"`
+}
+
+// Components is the struct filed of OpenAPISchema
+type Components struct {
+	Schemas Schemas `json:"schemas"`
+}
+
+// Schemas is the struct filed of Components
+type Schemas struct {
+	Parameter map[string]interface{} `json:"parameter"`
+}
+
+// GetDefinition gets OpenAPI schema from Cue section of a WorkloadDefinition/TraitDefinition
+func (s *APIServer) GetDefinition(c *gin.Context) {
+	definitionName := c.Param("name")
+	parameter, err := getDefinition(definitionName)
+	if err != nil {
+		util.HandleError(c, util.StatusInternalServerError, err)
```

Comment:
all errors are internal server error?

Reply:
Front-end devs only supported 200 and 500, so yes. But I will discuss with new front-end contributors whether we need more detailed error code and message.